### PR TITLE
BaSP-TG-113: Super admin form update

### PR DESCRIPTION
**Pull request by aletiscornia**
1. Replace the present logic for react hook form logic.
2. Add joi validations.

### CODE REVIEW

File: src/Components/SuperAdmins/Form/index.js [hunk 1]
```diff
@@ -41,17 +41,19 @@ const Form = () => {
           `${process.env.REACT_APP_API_URL}/superAdmins/${superAdminId}`
         );
         const data = await response.json();
```
Comment (MaicolFreire001) [inline src/Components/SuperAdmins/Form/index.js:43]: If we are goin to handle an async external call, we should handle it into the thunks.

File: src/Components/SuperAdmins/Form/index.js [hunk 2]
```diff
@@ -41,17 +41,19 @@ const Form = () => {
...
           password: data.data.password
-        });
+        };
         setTitle('Super admin edit');
+        reset(fetchData);
         return;
       } catch (error) {
         setTypeModal('Error');
-        setTextModal(error);
+        setTextModal('error');
```
Comment (MaicolFreire001) [inline src/Components/SuperAdmins/Form/index.js:56]: The error message should be more descriptive

**Review by MaicolFreire001 — CHANGES_REQUESTED** [review]

File: src/Components/SuperAdmins/Form/index.js (hunk 2 above)
Comment (aletiscornia) [reply src/Components/SuperAdmins/Form/index.js:56]: done

File: src/Components/SuperAdmins/Form/index.js (hunk 1 above)
Comment (aletiscornia) [reply src/Components/SuperAdmins/Form/index.js:43]: done

**Review by JuliMonte5 — APPROVED** [review]

**Review by MaicolFreire001 — APPROVED** [review]

**Review by sofihoj — APPROVED** [review]

**Pull request merged**